File: Integrated-Front/src (2)/src/app/advertisement.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { AdvertisementEntity } from './add-advertisement/AdvertisementEntity';
import { Advertisement } from './Advertisement';

@Injectable({
  providedIn: 'root'
})
export class AdvertisementService {

  constructor(private http:HttpClient) { }
  public url:string = "http://localhost:6060";
  public advertisementId:string;

  viewAdvertisements(userId:string, role:string):Observable<Advertisement[]>
  {
    return this.http.get<Advertisement[]>(this.url+'/ads/'+userId+'/'+role);
  }

  viewMyAdvertisements(userId:string):Observable<Advertisement[]>
  {
    return this.http.get<Advertisement[]>(this.url+'/myAds/'+userId);
  }
  addAdvertisement(ad:AdvertisementEntity):Observable<AdvertisementEntity>
  {
    return this.http.post<AdvertisementEntity>(this.url+"/addAdvertisement",ad);
  }

  deleteAdvertisement(advertisementId:string):Observable<string>
  {
    return this.http.delete<string>(this.url+'/deleteAdvertisement/'+advertisementId);
  }

  getAdvertisementId():Observable<string>
  {
    return this.http.get<string>(this.url+"/getId");
  }
  viewAd(adId:String):Observable<Advertisement>
  {
    return this.http.get<Advertisement>(this.url+'/getAd/'+adId);
  }

  
}

File: Integrated-Back/Backend1/Backend1/AdvertisementService (1)/AdvertisementService/src/main/java/com/cg/fas/service/AdvertisementService.java
package com.cg.fas.service;

import java.util.List;

import com.cg.fas.dto.AdvertisementDto;

public interface AdvertisementService {
	public AdvertisementDto addAdvertisement(AdvertisementDto advertisement);
	public String deleteAdvertisement(String advertisementId);
	public List<AdvertisementDto> viewMyAdvertisements(String userId);
	public List<AdvertisementDto> viewAdvertisements(String userId,String role);
	public String getLastAdvertisementId();
	public AdvertisementDto getAdvertisementById(String advertisementId);
	

}
